variables.tf

```diff
@@ -95,6 +95,56 @@ variable "managed_identities" {
   description = "Managed identities to be created for the resource."
 }
 
+variable "node_pools" {
+  type = map(object({
+    vm_size             = string
+    enable_auto_scaling = bool
+    max_count           = number
+    min_count           = number
+    node_count          = number
+    os_sku              = string
+    mode                = optional(string, "User")
+    os_disk_size_gb     = optional(number, null)
+    tags                = optional(map(string), {})
+    zones               = optional(string)
```

Comment:
Should this be a `set` of `string` like here ?
https://github.com/Azure/terraform-azurerm-aks/blob/4f7efad06d3471f22e960ad7874b2888e3754f26/variables.tf#L972